arangod/Replication/DatabaseInitialSyncer.cpp

```diff
@@ -534,28 +534,27 @@ Result DatabaseInitialSyncer::parseCollectionDumpMarker(transaction::Methods& tr
   }
 
   TRI_replication_operation_e type = REPLICATION_INVALID;
-  VPackSlice doc;
+  VPackSlice doc = marker;
+  VPackSlice key = doc.get(StaticStrings::KeyString);
 
-  for (auto it : VPackObjectIterator(marker, true)) {
-    if (it.key.isEqualString(kTypeString)) {
-      if (it.value.isNumber()) {
-        type = static_cast<TRI_replication_operation_e>(it.value.getNumber<int>());
-      }
-    } else if (it.key.isEqualString(kDataString)) {
-      if (it.value.isObject()) {
-        doc = it.value;
-      }
+  if (key.isString()) {
```

Comment:
sort of a brittle detection isn't it, shouldn't you check whether you set useEnvelope?
Also why not just dump vpack directly instead of using json? Should that not be even less traffic

Reply:
It is deliberately not using the value of `useEnvelope` here, because it is unknown if the leader supports it or not.
This allows an auto-detection of the format.
An alternative would be to make any leader that supports this format return a new response header, so that it can be detected without checking in each document.
Both will work.

Reply:
```suggestion
  if (useEnvelope && key.isString()) {
```

Reply:
why not set the accept header to use velocypack responses, should be even faster no?

Reply:
unless vpack validation is slower than JSON serialization / deserialization

Reply:
Done now.